src/test/java/de/doubleslash/keeptime/common/ConfigParserTest.java

```diff
@@ -0,0 +1,192 @@
+package de.doubleslash.keeptime.common;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.File;
+import java.util.Arrays;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import de.doubleslash.keeptime.controller.Controller;
+import de.doubleslash.keeptime.model.Model;
+import de.doubleslash.keeptime.model.Project;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.scene.paint.Color;
+
+public class ConfigParserTest {
+   private final Logger Log = LoggerFactory.getLogger(this.getClass());
+
+   ClassLoader classLoader = getClass().getClassLoader();
+   private final String CONFIG_FILENAME = classLoader.getResource("configTest.xml").getFile();
+
+   private final Project idle = new Project("Idle", Color.AQUAMARINE, true, 0);
+   private final Project heinz = new Project("Heinz", Color.BLUEVIOLET, true, 1);
+   private final Project peter = new Project("Peter", Color.CRIMSON, true, 2);
+   private final Project hinzUndKunz = new Project("Hinz und Kunz", Color.DARKRED, true, 3);
+   private final ObservableList<Project> projects = FXCollections.observableArrayList();
+
+   Model model = new Model();
+   private final ObservableList<Project> spyModel = spy(model.availableProjects);
```

Comment:
nice :) But as we mock the Controller, we don't have to test if the controller writes the new projects into the model correctly. We should just make sure, that controller.addNewProject is called as often as we expect with the needed parameters So we could get rid of all references of the model.